### PR TITLE
Add Debian trixie image

### PR DESCRIPTION
**Pull request by richlander**
It's another Debian release year! My assumption is that we'll start shipping Trixie images in Preview 1, in February. We should look at starting testing in January.

I was unable to make Arm32 work per https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/1286

I adopted a slightly different pattern, which I think is superior:

- One `apt-get updated` is consumed by all stages.
- Upstream image is mentioned once
- Removed `apt-get upgrade -y`, which seems wasteful.

### CODE REVIEW

**Comment by richlander** [comment]
I forgot that we don't have Trixie images in our mirror.

@lbussell

**Comment by mthalman** [comment]
> I forgot that we don't have Trixie images in our mirror.
> 
> @lbussell

Fixed

File: src/debian/manifest.json [hunk 1]
```diff
@@ -110,7 +110,34 @@
...
+                "debian-13-helix-amd64": {}
+              }
+            }
+          ]
+        },
+        {
+          "platforms": [
+            {
+              "architecture": "arm64",
+              "dockerfile": "src/debian/12/helix/",
```
Comment (mthalman) [inline src/debian/manifest.json:131]:
```suggestion
              "dockerfile": "src/debian/13/helix/",
```

**Review by mthalman — COMMENTED** [review]
> Removed apt-get upgrade -y, which seems wasteful.

Running an upgrade was intentional. See https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1226 for an explanation.

**Comment by richlander** [comment]
There is not. I was going to ping @rbhanda about that.

**Comment by richlander** [comment]
> Running an upgrade was intentional. 

I assumed. Why do we have a different practice for this repo than for dotnet/dotnet-docker? Is it that we have a different tradeoff on size vs security/compliance?

**Comment by rbhanda** [comment]
> There is not. I was going to ping @rbhanda about that.

https://github.com/dotnet/core/issues/9649

**Comment by mthalman** [comment]
> > Running an upgrade was intentional.
> 
> I assumed. Why do we have a different practice for this repo than for dotnet/dotnet-docker? Is it that we have a different tradeoff on size vs security/compliance?

Exactly. We don't have the same strictness in these images. It's ok if there's some wasted size in them if it makes our lives easier for maintaining them.

File: src/debian/13/helix/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,75 @@
+FROM library/debian:trixie AS base
+
+RUN apt-get update
```
Comment (MichaelSimons) [inline src/debian/13/helix/Dockerfile:3]: This adds bloat to the image.  It defeats the purpose of running `rm -rf /var/lib/apt/lists/*` later on.   Having a smaller image is more important that a faster image build.

**Comment by richlander** [comment]
@liveans -- I tested the latest `libmsquic` on Arm64 and it works there. It doesn't work on x64 for some reason.

I see this:

```bash
root@ad82881d09ec:/# dpkg -i libmsquic_2.4.7_amd64.deb 
Selecting previously unselected package libmsquic.
(Reading database ... 5854 files and directories currently installed.)
Preparing to unpack libmsquic_2.4.7_amd64.deb ...
Unpacking libmsquic (2.4.7) ...
dpkg: dependency problems prevent configuration of libmsquic:
 libmsquic depends on libxdp1; however:
  Package libxdp1 is not installed.
 libmsquic depends on libnl-route-3-200; however:
  Package libnl-route-3-200 is not installed.

dpkg: error processing package libmsquic (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libmsquic
```

**Comment by richlander** [comment]
Once the libmsquic issues are resolved, I'm going to do some quick size analysis based on the other feedback.

**Comment by liveans** [comment]
Can you also install these packages on x64?

`libnuma1 libelf1t64 libbpf1 libxdp1 libnl-3-200 libnl-route-3-200`

After than that, it should work

**Comment by richlander** [comment]
Done. I tested on x64 and Arm64 and that worked. I split the images into their own section so that they are easy to remove when we choose to install the libmsquic package.

**Comment by richlander** [comment]
This one passes! Thanks @liveans!

We're now down to more stylistic Dockerfile review.

File: src/debian/13/helix/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    && apt-get install -y \
+        cargo \
+        coreutils \
+        libffi-dev \
+        libssl-dev \
+        pkg-config \
+        python3-dev \
+        python3-pip \
+        python3-venv \
+        && rm -rf /var/lib/apt/lists/*
```
Comment (mthalman) [inline src/debian/13/helix/Dockerfile:13]:
```suggestion
    && rm -rf /var/lib/apt/lists/*
```

**Review by mthalman — APPROVED** [review]

**Pull request merged**